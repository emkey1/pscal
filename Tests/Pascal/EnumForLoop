program EnumForLoop;

type
  Day = (Mon, Tue, Wed, Thu, Fri, Sat, Sun);
  DaySet = set of Day;
  Person = record
    name: string;
    busy: DaySet;
  end;

var
  p: Person;
  d: Day;
begin
  p.name := 'Alice';
  p.busy := [Mon, Wed];
  for d := Mon to Sun do
    if d in p.busy then
      writeln(p.name, ' is busy on ', d);
end.
