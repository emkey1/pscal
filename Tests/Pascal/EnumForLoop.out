Alice is busy on mon
Alice is busy on wed
